src/inmanta/server/services/environment_metrics_service.py

```diff
@@ -360,11 +368,19 @@ async def get_metric_value(
         self, start_interval: datetime, end_interval: datetime, connection: asyncpg.connection.Connection
     ) -> Sequence[MetricValueTimer]:
         query: str = f"""
-            SELECT count(*) as count, environment, sum(completed-started) as compile_time
-            FROM {Compile.table_name()}
-            WHERE completed >= $1
-            AND completed < $2
-            GROUP BY environment
+            WITH compile_time as (
+                SELECT count(*) as count, environment, sum(completed-started)
+                FROM {Compile.table_name()}
+                WHERE completed >= $1
+                AND completed < $2
+                GROUP BY environment
+            ) SELECT e.id AS environment,
+            COALESCE(compile_time.count, 0) AS count,
+            COALESCE(compile_time.sum, INTERVAL '0' SECOND) as compile_time
```

Comment:
I think some indentation could help readability here.
```suggestion
            )
            SELECT e.id AS environment,
                COALESCE(compile_time.count, 0) AS count,
                COALESCE(compile_time.sum, INTERVAL '0' SECOND) AS compile_time
```